cdap-data-fabric/src/main/java/co/cask/cdap/data2/transaction/queue/AbstractQueueConsumer.java

```diff
@@ -137,19 +137,11 @@ public ConsumerConfig getConfig() {
   @Override
   public DequeueResult<byte[]> dequeue(int maxBatchSize) throws IOException {
     DequeueResult<byte[]> result = performDequeue(maxBatchSize);
-    if (scanStartRow != null) {
-      if (!consumingEntries.isEmpty()) {
-        // Start row can be updated to the largest rowKey in the consumingEntries (now is consumed)
-        // that is smaller than or equal to scanStartRow
-        byte[] floorKey = consumingEntries.floorKey(scanStartRow);
-        if (floorKey != null) {
-          updateStartRow(floorKey);
-        }
-      } else {
-        // If the dequeue has empty result, startRow can advance to scanStartRow
-        updateStartRow(scanStartRow);
-      }
-    }
+    // Start row can be updated to the largest rowKey in the consumingEntries
+    // that is smaller than or equal to scanStartRow. If no such key exists, update start row to scanStartRow
+    byte[] floorKey = consumingEntries.floorKey(scanStartRow);
```

Comment:
old code checks that scanStartRow isn't null but this doesn't.  Has it changed to never be null?